cpp/include/raft/handle.hpp

```diff
@@ -153,6 +137,15 @@ class handle_t {
     return cusparse_handle_;
   }
 
+  rmm::exec_policy get_thrust_policy() const {
+    std::lock_guard<std::mutex> _(mutex_);
+    if (!thrust_policy_initialized_) {
+      thrust_policy_ = new rmm::exec_policy(get_stream());
+      thrust_policy_initialized_ = true;
+    }
```

Comment:
Can we initialize in the constructor and ```set_stream``` and avoid this check? This function will be executed very frequently and I am not sure we should pay this cost every time.

Reply:
And we can avoid mutable variables in

https://github.com/rapidsai/raft/pull/286/files#diff-826899296b7bdbe71b6860ff89e5cad9c121f78e9dacfd0fded5ef217c04adefR232-R233

by doing so... In general, we'd better avoid mutable variables unless there is a very strong justification.

Reply:
Maybe @divyegala can be of good advice on this.

Reply:
I discussed with @divyegala about it offline. We agreed that it's indeed better to initialize the thrust execution policy during handler construction. @divyegala told me not to worry about the `set_stream` method as it will be removed soon in a following PR.